src/gluonnlp/utils/parameter.py

```diff
@@ -180,3 +277,66 @@ def move_to_ctx(arr, ctx):
         return [move_to_ctx(ele, ctx) for ele in arr]
     else:
         return None if arr is None else arr.as_in_ctx(ctx)
+
+
+def deduplicate_param_dict(param_dict):
+    """Get a parameter dict that has been deduplicated
+
+    Parameters
+    ----------
+    param_dict
+        The parameter dict returned by `model.collect_params()`
+
+    Returns
+    -------
+    dedup_param_dict
+    """
+    dedup_param_dict = dict()
+    param_uuid_set = set()
+    for k, v in param_dict.items():
```

Comment:
the key that will be kept in the resulting dictionary will not be deterministic because of the potentially different iteration order. I'd suggest only returning the value so that we won't run into inconsistent dictionary problem across workers due to incorrect assumption.

Reply:
an alternative would be to iterate with sorted keys.

Reply:
OK, I thought that starting from python3.6, this will be deterministic.

Reply:
Changed to use `sorted`.